services/basic_types.proto

```diff
@@ -1563,14 +1563,19 @@ message NftAllowance {
     /**
      * The list of serial numbers that the spender is permitted to transfer.
      */
-    repeated int64 serialNumbers = 4;
+    repeated int64 serial_numbers = 4;
+
+    /**
+     * If true, the spender has access to all of the owner's NFT units of type tokenId (currently
+     * owned and any in the future).
+     */
+    google.protobuf.BoolValue approved_for_all = 5;
 
     /**
-     * If true, the spender has access to all of the account owner's NFT instances (currently
-     * owned and any in the future). If this field is set to true the serialNumbers field
-     * should be empty.
+     * The account ID of the spender who is granted approvedForAll allowance and granting
+     * approval on an NFT serial to another spender.
      */
-    google.protobuf.BoolValue approvedForAll = 5;
+    AccountID delegating_spender = 6;
```

Comment:
`NftApproveAllowance{owner=Alice, spender=Carol, tokenId=FOOBAR, approvedForAll=false, serialNumbers=[1,2,3], delegatingSpender=Bob}`

This would be correct syntax to will let `Bob` give allowance to `Carol`. Mind that this txn has to be signed by `Bob` instead of owner `Alice` .

Bob can still spend the serials 1, 2 ,3.

Only approveForAll spenders can delegate. So there is only 1 level of such delegation.